pkg/receive/multitsdb.go

```diff
@@ -302,6 +310,9 @@ func (t *tenant) set(storeTSDB *store.TSDBStore, tenantTSDB *tsdb.DB, ship *ship
 }
 
 func (t *tenant) setComponents(storeTSDB *store.TSDBStore, ship *shipper.Shipper, exemplarsTSDB *exemplars.TSDB, tenantTSDB *tsdb.DB) {
+	if storeTSDB == nil && t.storeTSDB != nil {
+		t.storeTSDB.Close()
```

Comment:
This seems like strange, what does it do?

Reply:
This cancels cucko filter update loop for given tenant: https://github.com/thanos-io/thanos/pull/7787/files#diff-b3f73a54121d88de203946e84955da7027e3cfce7f0cd82580bf215ac57c02f4R121. I think it used in pruning TSDB in multi TSDB setup.

Without feature enabled it does nothing.